### PR TITLE
Updates the README file

### PR DESCRIPTION
**Pull request by psoloway**
1. Removed the "what this affects" section.
2. Made sure each parameter lists a data type and a default value. 
3. Made sure the links in the README work and are accurate. 
4. Updated the link in the Contributing section to point here: https://docs.puppetlabs.com/forge/contributing.html
5. General copyediting.
6. Removed testing info from the Limitations section; replaced it with a link to the matrix of supported platforms.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -143,127 +137,176 @@ The following parameters are available in the ntp module:
 
 ####`autoupdate`
 
-**Deprecated:** This parameter determined whether the ntp module should be
-automatically updated to the latest version available.  Replaced by `package_ensure`.
+**Deprecated; replaced by the `package_ensure` parameter**
+Tells Puppet whether to keep the ntp module updated to the latest version available.
+Valid options: 'true' or 'false'
+Default value: 'false'
```
Comment (igalic) [inline README.markdown]: are those supposed to be each on their own line in the rendered html?
cuz right now it ends up being on a single line.

Comment (psoloway) [inline README.markdown]: Thanks for the catch! Looks like one of my tools helpfully removed all my trailing spaces. I'll add them all back and submit another request...

Comment (psoloway) [inline README.markdown]: Correction! To maximize legibility while avoiding this happening again in future revisions, we've removed those line breaks and added periods instead. New PR incoming...

File: README.markdown [hunk 2]
```diff
@@ -139,145 +131,140 @@ class { '::ntp':
...
 
-Enables configs for undisciplined local clock, regardless of
-status as a virtual machine. 
-
+Specifies whether to enable specialized configuration options for an undisciplined local clock, regardless of its status as a virtual machine. Valid options: 'true' or 'false'. Default value: 'false'
 
 ##Limitations
 
-This module has been built on and tested against Puppet 3.
+This module has been tested on all PE-supported platforms, and no issues have been identified.
```
Comment (mhaskel) [inline README.markdown]: @psoloway ntp is not supported on all PE-supported platforms (https://forge.puppetlabs.com/puppetlabs/ntp/compatibility)...though I could just be reading this wrong?

Comment (laurenrother) [inline README.markdown]: @mhaskel @hunner What platforms is ntp supported on? The list is super long. Is it up-to-date? Do we want to keep it up to date? Is it enough to say that it is for sure supported on the OSes listed in the PE supported compatibility matrix https://forge.puppetlabs.com/supported ?

We're trying to standardize ALL the modules, and this is definitely not what's in the Limitations section for most modules. Should it be? Totally cool if so, I'm just curious.


Operating Systems:

* **RedHat**: 5, 6, 7
* **CentOS**: 5, 6, 7
* **OracleLinux**: 5, 6, 7
* **Scientific**: 5, 6, 7
* **SLES**: 10 SP4, 11 SP1, 12
* **Debian**: 6, 7
* **Ubuntu**: 10.04, 12.04, 14.04
* **Solaris**: 11
* **AIX**: 5.3, 6.1, 7.1


Operating Systems:

* **RedHat**: 5, 6, 7
* **CentOS**: 5, 6, 7
* **OracleLinux**: 5, 6, 7
* **Scientific**: 5, 6, 7
* **SLES**: 10 SP4, 11 SP1, 12
* **Debian**: 6, 7
* **Ubuntu**: 10.04, 12.04, 14.04
* **Solaris**: 11
* **AIX**: 5.3, 6.1, 7.1


Operating Systems:

* **RedHat**: 5, 6, 7
* **CentOS**: 5, 6, 7
* **OracleLinux**: 5, 6, 7
* **Scientific**: 5, 6, 7
* **SLES**: 10 SP4, 11 SP1, 12
* **Debian**: 6, 7
* **Ubuntu**: 10.04, 12.04, 14.04
* **Solaris**: 11
* **AIX**: 5.3, 6.1, 7.1


Operating Systems:

* **RedHat**: 5, 6, 7
* **CentOS**: 5, 6, 7
* **OracleLinux**: 5, 6, 7
* **Scientific**: 5, 6, 7
* **SLES**: 10 SP4, 11 SP1, 12
* **Debian**: 6, 7
* **Ubuntu**: 10.04, 12.04, 14.04
* **Solaris**: 11
* **AIX**: 5.3, 6.1, 7.1


Operating Systems:

* **RedHat**: 5, 6, 7
* **CentOS**: 5, 6, 7
* **OracleLinux**: 5, 6, 7
* **Scientific**: 5, 6, 7
* **SLES**: 10 SP4, 11 SP1, 12
* **Debian**: 6, 7
* **Ubuntu**: 10.04, 12.04, 14.04
* **Solaris**: 11
* **AIX**: 5.3, 6.1, 7.1


Operating Systems:

* **RedHat**: 5, 6, 7
* **CentOS**: 5, 6, 7
* **OracleLinux**: 5, 6, 7
* **Scientific**: 5, 6, 7
* **SLES**: 10 SP4, 11 SP1, 12
* **Debian**: 6, 7
* **Ubuntu**: 10.04, 12.04, 14.04
* **Solaris**: 11
* **AIX**: 5.3, 6.1, 7.1

**Comment by mhaskel** [comment]
@psoloway @laurenrother I think supported on OSes listed in the compatibility matrix is good!

I don't think this should be in the limitations section for most/all modules.

**Comment by laurenrother** [comment]
@mhaskel Cool, so the original wording should be good, but there should be a link pointing to the Supported page (since there's apparently no direct link to the the matrix(?!) [You can tell I just wrote the copy and didn't design the page])?

**Comment by mhaskel** [comment]
@laurenrother Yeah, I think if there's a link it should be clear. Just on initial reading it sounded like it worked on _all_ PE platforms, so I got confused :)

**Comment by psoloway** [comment]
@mhaskel @laurenrother For what it's worth, there's supposed to be a direct link to the matrix and I have a ticket in to fix it. Haven't gotten any love yet, though. https://tickets.puppetlabs.com/browse/PF-1030

**Comment by psoloway** [comment]
@mhaskel Okay, I added the link and left the copy unchanged. The link points to the anchor that's supposed to exist at the top of the matrix, so whenever they get around to fixing it our link should work too. In the meantime, it'll still get folks as far as the top of the page. Otherwise good to merge?

**Comment by mhaskel** [comment]
@psoloway looks good, but could you squash this down to a single commit? (I can help you with that if you need it)

**Comment by psoloway** [comment]
Git has gotten the best of me and I'm going to submit the new pull request from a different fork. I know what I did wrong... or at least how to do it right the next time.

**Comment by elyscape** [comment]
For future reference, you can squash/fixup several commits down into one by using `git rebase`.  The technique is described [in this article](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).  Once you've squashed the commits, you can push them back up to GitHub with `git push --force`.

**Comment by psoloway** [comment]
Yeah I found that out only after I'd irreparably screwed with my repo (`git rebase` currently cause some kind of loop and never completes.) I'll do it the right way next time, I promise :)

**Pull request closed without merging**